cmd/sync-tags/main.go

```diff
@@ -258,71 +279,69 @@ func main() {
 		// update go.mod or Godeps.json to point to actual tagged version in the dependencies. This version might differ
 		// from the one currently in go.mod  or Godeps.json because the other repo could have gotten more commit for this
 		// tag, but this repo didn't. Compare https://github.com/kubernetes/publishing-bot/issues/12 for details.
+		var changed, goModExists bool
+		_, err = os.Stat("go.mod")
+		if err == nil {
+			goModExists = true
+		}
+
 		if len(dependentRepos) > 0 {
-			fmt.Printf("Checking that dependencies point to the actual tags in %s.\n", strings.Join(dependentRepos, ", "))
-			wt, err := r.Worktree()
-			if err != nil {
-				glog.Fatalf("Failed to get working tree: %v", err)
-			}
-			fmt.Printf("Checking out branch tag commit %s.\n", bh.String())
-			if err := wt.Checkout(&gogit.CheckoutOptions{Hash: bh}); err != nil {
-				glog.Fatalf("Failed to checkout %v: %v", bh, err)
-			}
+			wt := checkoutBranchTagCommit(r, bh, dependentRepos)
 
 			// if go.mod exists, fix only go.mod and generate Godeps.json from it later
 			// if it doesn't exist, check if Godeps.json exists, and update it
-			var changed, goModChanged bool
-			_, err = os.Stat("go.mod")
-			if os.IsNotExist(err) {
+			if !goModExists {
 				if _, err2 := os.Stat("Godeps/Godeps.json"); err2 == nil {
 					fmt.Printf("Updating Godeps.json to point to %s tag.\n", bName)
 					changed, err = updateGodepsJsonWithTaggedDependencies(bName, dependentRepos)
 					if err != nil {
 						glog.Fatalf("Failed to update Godeps.json for tag %s: %v", bName, err)
 					}
 				}
-			} else if err == nil {
-				fmt.Printf("Updating go.mod and go.sum to point to %s tag.\n", bName)
-				changed, err = updateGomodWithTaggedDependencies(bName, dependentRepos)
-				if err != nil {
-					glog.Fatalf("Failed to update go.mod and go.sum for tag %s: %v", bName, err)
-				}
-				goModChanged = true
-			}
-
-			if goModChanged && *generateGodeps {
-				fmt.Printf("Regenerating Godeps.json from go.mod.\n")
-				if err := regenerateGodepsFromGoMod(); err != nil {
-					glog.Fatalf("Failed to regenerate Godeps.json from go.mod: %v", err)
+			} else {
+				if publishSemverTag {
+					changed = updateGoModAndGodeps(semverTag, dependentRepos, *generateGodeps, true)
+				} else {
+					changed = updateGoModAndGodeps(bName, dependentRepos, *generateGodeps, false)
```

Comment:
does this mean we don't support semver tags and `kubernetes-*` tags in parallel?

Reply:
> does this mean we don't support semver tags and kubernetes-* tags in parallel?

We do. Both of them point to the same sha, so we create tags at the sha later (like you commented below).

Reply:
annotated tags lead to different shas. Doesn't this mean conflicting shas if I pin all dependencies to kubernetes-v1.17.0, but e.g. client-go kuberentes-v1.17.0 points to apimachinery v0.17.0 which is different from apimachiner kubernetes-v1.17.0 ?

Reply:
> annotated tags lead to different shas

really? https://github.com/prowtest/client-go/releases shows identical shas for kubernetes-1.17.0-rc.2 and v0.17.0-rc.2